src/Symfony/Component/HttpFoundation/Tests/StreamedJsonResponseTest.php

```diff
@@ -0,0 +1,223 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpFoundation\Tests;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\HttpFoundation\StreamedJsonResponse;
+
+class StreamedJsonResponseTest extends TestCase
+{
+    public function testResponseSimpleList()
+    {
+        $content = $this->createSendResponse(
+            [
+                '_embedded' => [
+                    'articles' => $this->generatorSimple('Article'),
+                    'news' => $this->generatorSimple('News'),
+                ],
+            ],
+        );
+
+        $this->assertSame('{"_embedded":{"articles":["Article 1","Article 2","Article 3"],"news":["News 1","News 2","News 3"]}}', $content);
+    }
+
+    public function testResponseObjectsList()
+    {
+        $content = $this->createSendResponse(
+            [
+                '_embedded' => [
+                    'articles' => $this->generatorArray('Article'),
+                ],
+            ],
+        );
+
+        $this->assertSame('{"_embedded":{"articles":[{"title":"Article 1"},{"title":"Article 2"},{"title":"Article 3"}]}}', $content);
+    }
+
+    public function testResponseWithoutGenerator()
+    {
+        // while it is not the intended usage, all kind of iterables should be supported for good DX
+        $content = $this->createSendResponse(
+            [
+                '_embedded' => [
+                    'articles' => ['Article 1', 'Article 2', 'Article 3'],
+                ],
+            ],
+        );
+
+        $this->assertSame('{"_embedded":{"articles":["Article 1","Article 2","Article 3"]}}', $content);
```

Comment:
maybe we should write tests in this way to ease readability and be sure that the output is really a valid json_encode output

```suggestion
        // while it is not the intended usage, all kind of iterables should be supported for good DX
        $payload = [
            '_embedded' => [
                'articles' => ['Article 1', 'Article 2', 'Article 3'],
            ],
        ];
        
        $content = $this->createSendResponse(payload);

        $this->assertSame(json_encode(payload), $content);
```

Reply:
@jderusse that will not work as most of the tests there is no 1to1 mapping for json_encode. As json_encode does not handle Generators if would would handle that we would not need this class: https://3v4l.org/83O1p